cargo-up/cargo-up-derive/src/impls/upgrader.rs

```diff
@@ -23,28 +18,13 @@ pub fn upgrader(input: &DeriveInput) -> TokenStream {
         _ => abort_call_site!("`#[upgrader]` only supports non-tuple structs"),
     };
 
-    fields.push(Field {
-        attrs: vec![],
-        vis: Visibility::Inherited,
-        ident: Some(parse_str("_upgrader").unwrap()),
-        colon_token: Some(Colon::default()),
-        ty: parse_str("::cargo_up::UpgraderInner").unwrap(),
-    });
-
-    let mut new = input.clone();
-
-    if let Data::Struct(d) = new.data {
-        new.data = Data::Struct(DataStruct {
-            fields: Fields::Named(FieldsNamed {
-                brace_token: Brace::default(),
-                named: fields,
-            }),
-            ..d
-        });
-    }
+    let upgrader = quote!(_upgrader: ::cargo_up::UpgraderInner);
 
     quote! {
-        #new
+        struct #ident {
```

Comment:
We are missing `DeriveInput`'s, `attrs`, `vis`, `generics`. I remember I didn't do it like this because  tokenising `generics` was a bit difficult because you have to put it's types at one point and it's where clause at other.

Reply:
I definitely like the PR, but this is a blocker.